mysys/CMakeLists.txt

```diff
@@ -79,9 +79,14 @@ ELSEIF(CMAKE_SYSTEM_PROCESSOR MATCHES "x86_64|amd64|i386|i686")
     ENDIF()
   ENDIF()
 ELSEIF(CMAKE_SYSTEM_PROCESSOR MATCHES "aarch64|AARCH64")
-  IF(CMAKE_C_COMPILER_ID STREQUAL "GNU")
+  IF(CMAKE_C_COMPILER_ID MATCHES "Clang|GNU")
     include(CheckCXXSourceCompiles)
 
+    CHECK_C_COMPILER_FLAG(-march=armv8-a+crc+crypto HAVE_ARMV8_CRC_CRYPTO_MARCH)
+    IF(HAVE_ARMV8_CRC_CRYPTO_MARCH)
+      SET(CMAKE_REQUIRED_FLAGS "-march=armv8-a+crc+crypto")
```

Comment:
please do a save CMAKE_REQUIRED_FLAGS, *append* to CMAKE_REQUIRED_FLAGS, and a restoration in the final block.

There's a few violation of this behaviour currently I know, but the auth plugins are self contained to the small number of files, configure.cmake might be occurring just early enough for any other flags to be omitted and does the SunOS.cmake. `build_rocksdb.cmake` will probably need fixing because it might override other settings for its `HAVE_SSE42`/`HAVE_PCLMUL` test.